actionpack/lib/action_controller/metal/strong_parameters.rb

```diff
@@ -621,17 +621,7 @@ def init_with(coder) # :nodoc:
 
     def method_missing(method_sym, *args, &block)
       if @parameters.respond_to?(method_sym)
-        message = <<-DEPRECATE.squish
-          Method #{method_sym} is deprecated and will be removed in Rails 5.1,
-          as `ActionController::Parameters` no longer inherits from
-          hash. Using this deprecated behavior exposes potential security
-          problems. If you continue to use this method you may be creating
-          a security vulnerability in your app that can be exploited. Instead,
-          consider using one of these documented methods which are not
-          deprecated: http://api.rubyonrails.org/v#{ActionPack.version}/classes/ActionController/Parameters.html
-        DEPRECATE
-        ActiveSupport::Deprecation.warn(message)
-        @parameters.public_send(method_sym, *args, &block)
+        to_h.public_send(method_sym, *args, &block)
```

Comment:
@rafaelfranca may I ask why you would prefer to keep the deprecation?  The existence of `#to_hash` on the class suggests that instances may stand in as instances of Hash -- which seems appropriate.  By forwarding to the result of `#to_h` more existing code will continue to work, with no loss of security.